Assets/Scripts/TrapDoorOpen.cs

```diff
@@ -4,14 +4,23 @@
 
 public class TrapDoorOpen : MonoBehaviour
 {
-    public GameObject TrapDoor;
+    [SerializeField]
+    private GameObject TrapDoor;
+    
+    public delegate void TrapOpenedHandler();
+    public event TrapOpenedHandler TrapOpened;
     
     void OnTriggerEnter(Collider other){
         if(other.gameObject.tag == "Player"){
-        TrapDoor.transform.GetChild(0).GetComponent<Animation>().Play("LeftAnim");
-        TrapDoor.transform.GetChild(1).GetComponent<Animation>().Play("HingeRight");
+            //Play animations
+            TrapDoor.transform.GetChild(0).GetComponent<Animation>().Play("LeftAnim");
+            TrapDoor.transform.GetChild(1).GetComponent<Animation>().Play("HingeRight");
+            
+            //signal event
+            TrapOpened?.Invoke();

```

Comment:
I can see this event being useful for something that is unrelated to the trapdoor or the player (Such as the HUD). But why are you using this event to then pull the player, when you have the player right here, in this method, as `Collider other`??? 

Now that I look at it, I can't tell where you placed that `PlayerFall.cs` component. I hope you didn't just place on some random, unrelated gameobject. If you didn't, then why not use the gameobject you place it on to it's advantage (Seems like you are "Find"ing references to the everything in that file).

Reply:
Placed on the TrapDoor parent object, the hinges contain TrapDoorOpen.cs components.